cocotb/share/lib/simulator/simulatormodule.cpp

```diff
@@ -923,62 +923,138 @@ PyMODINIT_FUNC PyInit_simulator(void)
+            "--\n\n"
+            "Set the value of a signal using an NUL-terminated 8-bit string"
+        )
+    },
     {"set_signal_val_binstr",
-        (PyCFunction)set_signal_val_binstr, METH_VARARGS,
-        "Set the value of a signal using a string with a character per bit"},
+        (PyCFunction)set_signal_val_binstr, METH_VARARGS, PyDoc_STR(
+            "set_signal_val_binstr($self, action, value, /)\n"
+            "--\n\n"
+            "Set the value of a signal using a string with a character per bit"
+        )
+    },
     {"set_signal_val_real",
-        (PyCFunction)set_signal_val_real, METH_VARARGS,
-        "Set the value of a signal using a double precision float"},
+        (PyCFunction)set_signal_val_real, METH_VARARGS, PyDoc_STR(
+            "set_signal_val_real($self, action, value, /)\n"
+            "--\n\n"
+            "Set the value of a signal using a double precision float"
+        )
+    },
     {"get_definition_name",
-        (PyCFunction)get_definition_name, METH_NOARGS,
-        "Get the name of a GPI object's definition"},
+        (PyCFunction)get_definition_name, METH_NOARGS, PyDoc_STR(
+            "get_definition_name($self)\n"
+            "--\n\n"
+            "Get the name of a GPI object's definition"
+        )
+    },
     {"get_definition_file",
-        (PyCFunction)get_definition_file, METH_NOARGS,
-        "Get the file that sources the object's definition"},
+        (PyCFunction)get_definition_file, METH_NOARGS, PyDoc_STR(
+            "get_definition_file($self)\n"
+            "--\n\n"
+            "Get the file that sources the object's definition"
+        )
+    },
     {"get_handle_by_name",
-        (PyCFunction)get_handle_by_name, METH_VARARGS,
-        "Get handle of a named object"},
+        (PyCFunction)get_handle_by_name, METH_VARARGS, PyDoc_STR(
+            "get_handle_by_name($self, name, /)\n"
+            "--\n\n"
+            "Get handle of a named object"
+        )
+    },
     {"get_handle_by_index",
-        (PyCFunction)get_handle_by_index, METH_VARARGS,
-        "Get handle of a object at an index in a parent"},
+        (PyCFunction)get_handle_by_index, METH_VARARGS, PyDoc_STR(
+            "get_handle_by_index($self, index, /)\n"
+            "--\n\n"
+            "Get handle of a object at an index in a parent"
+        )
+    },
     {"get_name_string",
-        (PyCFunction)get_name_string, METH_NOARGS,
-        "Get the name of an object as a string"},
+        (PyCFunction)get_name_string, METH_NOARGS, PyDoc_STR(
+            "get_name_string($self)\n"
+            "--\n\n"
+            "Get the name of an object as a string"
+        )
+    },
     {"get_type_string",
-        (PyCFunction)get_type_string, METH_NOARGS,
-        "Get the type of an object as a string"},
+        (PyCFunction)get_type_string, METH_NOARGS, PyDoc_STR(
+            "get_type_string($self)\n"
+            "--\n\n"
+            "Get the type of an object as a string"
```

Comment:
```suggestion
            "Get the GPI enumeration type of an object as a string"
```

This got me the first time I used it. I thought it was going to be useful...